starknet_py/net/full_node_client.py

```diff
@@ -474,11 +473,32 @@ def _create_broadcasted_txn_common_properties(transaction: AccountTransaction) -
         "type": "INVOKE"
         if transaction.tx_type == TransactionType.INVOKE_FUNCTION
         else transaction.tx_type.name,
-        "max_fee": hash_to_felt(transaction.max_fee),
-        "version": hash_to_felt(transaction.version),
-        "signature": [hash_to_felt(sig) for sig in transaction.signature],
-        "nonce": hash_to_felt(transaction.nonce)
+        "max_fee": _to_rpc_felt(transaction.max_fee),
+        "version": _to_rpc_felt(transaction.version),
+        "signature": [_to_rpc_felt(sig) for sig in transaction.signature],
+        "nonce": _to_rpc_felt(transaction.nonce)
         if transaction.nonce is not None
-        else None,
+        else "0x00",
     }
     return broadcasted_txn_common_properties
+
+
+def _to_storage_key(key: int) -> str:
+    hashed_key = hex(key).lstrip("0x")
+
+    if hashed_key[0] not in ("0", "1", "2", "3", "4", "5", "6", "7"):
```

Comment:
```suggestion
    if hashed_key[0] not in str(range(8)):
```
?

Reply:
That won't work, it'd evaluate to `Range(0, 8)` or something like this. We could do `[str(i) for i in range(8)]` if you think that's better?

Reply:
Maybe change it to just `0 <= int(x) <= 7`

Reply:
Oh, it could be `a - f` as well

Reply:
> Maybe change it to just `0 <= int(x) <= 7`

That'd have to be `0 <= int(hashed_key[0], 16) <= 7`. Is that more readable?

Reply:
To be fair, the original code is more readable 🙈

Reply:
Let's keep it then.